### PR TITLE
reduce too verbose logging sharding, #27276

### PR DESCRIPTION
**Pull request by patriknw**
Such as:
* the forward message log in the region
* logging of full state in the coordinator

References #27276

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala [hunk 1]
```diff
@@ -525,7 +525,7 @@ object ShardCoordinator {
     }
     var remaining = regions
 
-    log.debug("Rebalance [{}] from region [{}]", shard, regions)
+    log.debug("Rebalance [{}] from [{}] regions", regions.size, shard)
```
Comment (renatocaval) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala:528]: The args got swapped. Was that intentional? 
We are rebalancing the shard (1st arg) from region(s) (2nd arg). 

Instead we could have the following:

```suggestion
    log.debug("Rebalance [{}] from [{}] region(s)", shard, regions.size)
```

Sometimes we have only one region hence the `(s)`. 

Moreover, I don't think we should say "rebalance" anymore. What is really happening here is a hand-off. We are asking the region to forget the location of the shard and if the region is hosting it, we are also asking the region to let it go (stop the shard). 

This is often done when rebalancing, but also when just removing the last region. On a side node, the `RebalanceWorker` could be renamed to `HandOffWorker`.

Anyway, I was planning to change the wording on the working I'm doing for #29549

**Review by octonato — APPROVED** [review]
LGTM

See my comment about #29549. Part of the issue is how we are logging things.

**Review by johanandren — APPROVED** [review]

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala (hunk 1 above)
Comment (patriknw) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala:528]: right, wrong order of the parameters

**Comment by renatocaval** [comment]
This was merged with the wrong order of parameters. I will fix it in #29579 since I'm touching the logs already.

**Pull request merged**